src/core/router.js

```diff
@@ -231,9 +233,16 @@ var router = {
                         window.require(
                             dependencies,
                             function() {
+                                if(_.has(self.activeController, 'destroy')){
+                                    self.activeController.destroy();
+                                }
+
+                                self.trigger('destroy');
+
                                 _.forEach(arguments, function(dependency) {
                                     if (dependency && _.isFunction(dependency.start)) {
                                         dependency.start();
+                                        self.activeController = dependency;
```

Comment:
As we now have a `destroy` event, we could also have a `start` event.
```suggestion
                                        activeController = dependency;
                                        self.trigger('start', activeController)
```

Reply:
It's exactly what I wanted to do next https://github.com/oat-sa/extension-tao-mpart/pull/814/files?file-filters%5B%5D=#diff-a69fe564ddfd4128784c4b7dd3ced954R363 :)